readium/adapters/exoplayer/audio/src/main/java/org/readium/adapter/exoplayer/audio/ExoPlayerEngineProvider.kt

```diff
@@ -77,6 +82,22 @@ public class ExoPlayerEngineProvider(
         return Try.success(engine)
     }
 
+    private fun createDataSourceFactory(
+        publication: Publication
+    ): DataSource.Factory {
+        val baseDataSource = ExoPlayerDataSource.Factory(publication)
+        val cache = cacheProvider?.getCache(publication)
+            ?: return baseDataSource
+
+        return CacheDataSource.Factory()
+            .setCache(cache)
+            .setCacheKeyFactory { cacheProvider.computeKey(publication, Href(it.uri.toUrl()!!)) }
```

Comment:
Why pass an `Href` instead of an `Url`, if we have to compute it from a static URL every time?

Reply:
I thought the `Href` class conveyed better that what need is a resource identifier. Unfortunately, `Href` can also be templated. I'll change that.